### DISCUSSION TITLE
How can I safely remove specific shadCN UI components from my Next.js project??

### DISCUSSION THREAD

**Discussion by bipanshu-kumar** [community/community#143976] [Programming Help] [answered]
I'm using ShadCN components in a Next.js project, and I need to remove some UI components that are no longer necessary. I want to make sure they’re removed cleanly, without leaving any unused imports, dependencies, or configurations that might affect app performance or cause errors.

Here’s what I’d like help with:

1. After deleting the component files from my `/components/ui` folder, are there any additional steps I should take to ensure no unused imports or configuration are left behind?
   
2. Are there any dependencies that could be safely uninstalled? And are there any recommended tools to help identify unused dependencies?

Any tips on best practices for cleanly removing components in a Next.js project using ShadCN and Tailwind CSS would be really helpful!

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @bipanshu-kumar!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!

**Comment by jgschmitz** [comment, marked as answer]
# Cleanly Removing UI Components in a Next.js Project with ShadCN and Tailwind CSS

This guide outlines the steps to cleanly remove UI components in a Next.js project, ensuring no unused imports, dependencies, or configurations remain.

---

## 1. Remove Component Files
- Delete the component files from the `/components/ui` folder or their respective location.
- Verify that no other components or pages are referencing the deleted files. Next.js will throw build-time errors if there are unresolved imports.

---

## 2. Search for Unused Imports
- Use your code editor’s **"Find All References"** or global search to locate where the deleted component was imported.
- Remove any unused imports in the files where the component was previously used.

For example:
```jsx
import { UnusedComponent } from "@/components/ui";

**Reply by harshG775** [reply]
what about the dependencies/devDependencies in package.json

**Reply by exodus-tola-mindCoder** [reply]
yoo this is a solution


https://www.npmjs.com/package/shadcn-remover

I was working on a Next.js project with Shadcn and needed to delete some components.
Checked the docs, googled it no easy way, just manual work.
That’s when the idea hit me why not build a tool for it? So I did mainly for myself!

Check out my repo

https://github.com/exodus-tola-mindCoder/shadcn-remover

**Reply by tolex118** [reply]
hey this is not spam why you guys marks as a spam message?
personally l liked it and the project is very useful.

**Reply by humayunkabir** [reply]
@exodus-tola-mindCoder, Does this [shadcn-remover](https://www.npmjs.com/package/shadcn-remover) package also remove the dependencies safely? And before removing, does it check the project folder if there are any imports in any files?

**Comment by DevJogger** [comment]
I ran into a similar issue recently. What I ended up doing was checking the code of each specific component under /components/ui to understand its dependencies, and then manually removing the unused dependencies/devDependencies from package.json.

I plan to do this from now on, using `git commit` every time while installing a shadcn/ui component, with a commit message indicating which component was added. While I need to uninstall a component, I can simply use `git revert` to roll back the related code and dependencies cleanly. I haven’t tried this approach yet, but it seems like a reliable way to manage component changes.

**Reply by exodus-tola-mindCoder** [reply]
Yoo here is the solution

https://www.npmjs.com/package/shadcn-remover 

I was working on a Next.js project with Shadcn and needed to delete some components.
Checked the docs, googled it no easy way, just manual work.
That’s when the idea hit me why not build a tool for it? So I did mainly for myself!

Check my repo
https://github.com/exodus-tola-mindCoder/shadcn-remover

**Comment by exodus-tola-mindCoder** [comment]
https://www.npmjs.com/package/shadcn-remover

I was working on a Next.js project with Shadcn and needed to delete some components.
Checked the docs, googled it no easy way, just manual work.
That’s when the idea hit me why not build a tool for it? So I did mainly for myself! 

check out my repo
https://github.com/exodus-tola-mindCoder/shadcn-remover

**Comment by mecodeatlas** [comment]
Hey @bipanshu-kumar! We're going through discussions that haven't been marked as answered yet. Did the replies from other members help you resolve the issue you raised? If so, can you please mark their response as the answer? When you mark a question as an answer, GitHub will highlight the comment and replies to the comment to help visitors quickly find the answer.
![image](https://github.com/user-attachments/assets/b89e7161-c2d1-47e2-a058-242df633e6b2)

**Comment by tolex118** [comment]
> I'm using ShadCN components in a Next.js project, and I need to remove some UI components that are no longer necessary. I want to make sure they’re removed cleanly, without leaving any unused imports, dependencies, or configurations that might affect app performance or cause errors.
> 
> Here’s what I’d like help with:
> 
> 1. After deleting the component files from my `/components/ui` folder, are there any additional steps I should take to ensure no unused imports or configuration are left behind?
> 2. Are there any dependencies that could be safely uninstalled? And are there any recommended tools to help identify unused dependencies?
> 
> Any tips on best practices for cleanly removing components in a Next.js project using ShadCN and Tailwind CSS would be really helpful!

Introducing: shadcn-remover CLI
A simple, powerful, and user-friendly CLI tool to efficiently remove Shadcn UI components from your project.

While working on a Next.js project with Shadcn, I added a lot of UI components. Later wanted to remove some of them, I realized there was no easy way to do it.
I checked the official Shadcn docs, browsed through their GitHub repo, and even Googled around — but found no dedicated command or simple solution. Manual deletion was the only option.
That’s when the idea hit me: Why not build a tool for that? — and shadcn-remover CLI was born.

It automates the cleanup process and comes packed with useful features.
By default, it expects your Shadcn UI components to be located at src/components/ui/ (the standard Shadcn structure in  Next.jsV13++).

Check out the full feature list on the GitHub repo and the package README.

Your support through stars and shares is greatly appreciated!

Thanks for using and sharing the shadcn-remover CLI!

NPM: https://www.npmjs.com/package/shadcn-remover
GitHub: https://github.com/exodus-tola-mindCoder/shadcn-remover

**Reply by Exodus-tola** [reply]
this is cool and useful project  thank you for sharing and keep it up!

**Comment by Exodus-tola** [comment]
Yes sir!
